doc/source/_toc.yml

```diff
@@ -66,7 +66,12 @@ parts:
           - file: serve/core-apis
           - file: serve/http-servehandle
           - file: serve/ml-models
-          - file: serve/deployment-graph
+          - file: serve/deployment-graph/deployment-graph
+            sections:
+              - file: serve/deployment-graph/deployment-graph-e2e-tutorial
```

Comment:
you know I'm against this "feature-first" approach to docs hierarchy, but I can't stop you guys from doing your thing.

Reply:
We're referencing Ray dataset structure which i believe you contributed a lot to its format, open to changes :)  iiuc you're suggesting we should bring that to first level user guide instead? I think that's better format too to avoid nesting docs, but meanwhile Deployment graph was just in alpha stage but i haven't found a structure that satisfies both yet. What do you think is more reasonable?

Reply:
Thanks for the inputs Max, it makes a lot of sense. As we chatted earlier in March, Serve team should re-structure our docs to better organize our content on this and it's happening now, please don't hesitate to step up and give us suggestions like this :)

For this PR in particular, Sihan is ramping on the context to make progress, but I will self-assign and ensure syncing with Simon to properly organize them as part of the docs restructure effort.